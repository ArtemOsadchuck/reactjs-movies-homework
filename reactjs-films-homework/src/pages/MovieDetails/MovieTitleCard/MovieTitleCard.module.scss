$background-card-color: rgb(0, 0, 0);
$background-img-color: rgb(231, 231, 231);
$background-rating-color: rgb(41, 186, 41);
$play-icon-size: 44px;
$wite-color: rgb(255, 255, 255);

.movieTitleWrapper {
  display: flex;
  height: max-content;
  margin: 0;
  min-height: 22rem;
  overflow: hidden;
  position: relative;
  width: 100%;

  @media screen and (max-width: 740px) {
    flex-wrap: wrap;
    margin: 0;
  }

  .imgCardWrapper {
    align-items: center;
    background-color: $background-img-color;
    border-radius: 5px;
    display: flex;
    flex-wrap: wrap;
    height: 29rem;
    justify-content: center;
    margin: 5% auto;
    transition: 0.5s ease-in-out;
    width: 90%;
  }

  h4 {
    font-weight: 400;
    letter-spacing: 1px;
    padding-left: 5%;
    text-align: start;
  }

  .ratingMovie {
    align-items: center;
    background-color: $background-rating-color;
    border-radius: 3px;
    color: $wite-color;
    display: flex;
    height: 1.3rem;
    justify-content: center;
    left: 1rem;
    overflow: hidden;
    position: absolute;
    top: 1.5rem;
    width: 2rem;
    z-index: 11;

    .rating {
      font-size: 0.8rem;
      opacity: 1;
    }
  }

  .movieCardWrapper {
    align-items: center;
    border-radius: 2px;
    color: $background-img-color;
    display: block;
    flex-shrink: 0;
    letter-spacing: 1px;
    min-height: 30rem;
    overflow: hidden;
    padding: 0 5px;
    position: relative;
    text-align: start;
    width: 22rem;
    z-index: 10;
  }

  .genreWrapper {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    min-width: 11rem;
    width: fit-content;
  }

  .genre {
    align-items: center;
    background-color: $background-rating-color;
    border-radius: 5px;
    display: flex;
    font-size: 1rem;
    height: 1.5rem;
    justify-content: center;
    min-width: 5rem;
    text-align: center;
    width: max-content;

    &:hover {
      border-radius: 10px;
      cursor: pointer;
      letter-spacing: 1px;
      transition: 0.2s ease-in-out;
    }
  }

  .infoWrapper {
    color: $wite-color;
    display: flex;
    flex-direction: column;
    height: 29rem;
    justify-content: space-between;
    margin-right: 1rem;
    margin-top: 1rem;
    max-width: 40rem;
    padding-bottom: 2rem;
    text-align: start;

    @media screen and (max-width: 740px) {
      margin: 0 auto;
      padding: 0 1.5rem;
    }
  }
}
