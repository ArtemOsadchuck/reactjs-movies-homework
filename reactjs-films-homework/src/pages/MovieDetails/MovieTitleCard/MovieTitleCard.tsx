import React from 'react';
import img from '../../../components/MovieCard/img/Movie-card.png';
import styles from './MovieTitleCard.module.scss';
import MovieTitleInfo from './MovieTitleInfo';

import lang from '../../../languages/getLanguage';
import { useAppSelector } from '../../../hooks/hooks';

import getRevenueFormat from './utils/getRevenueFormat';

export interface ICard {
  props: ITitleMovieProps;
}
export interface ITitleMovieProps {
  id: number;
  title: string;
  vote_average: number;
  poster_path: string;
  overview: string;
  release_date: string;
  runtime: number;
  revenue: number;
  genres?: {
    id: number;
    name: string;
  }[];
}

const MovieTitleCard: React.FC<ICard> = ({ props }) => {
  const {
    id,
    title,
    vote_average,
    poster_path,
    overview,
    release_date,
    genres,
    runtime,
    revenue,
  } = props;
  const urlImg = `https://image.tmdb.org/t/p/w500/${poster_path}`;
  const imgWidth = '52px';
  const appLang = useAppSelector((state) => state.mainReducer.lang);
  const titleText = lang(appLang).titleName + ':';
  const overviewText = lang(appLang).overview + ':';
  const releaseText = lang(appLang).releaseDate + ':';
  const revenueText = lang(appLang).revenue + ':';
  const durationText = lang(appLang).duration + ':';
  const durationTimeMin = lang(appLang).min;

  return (
    <div id={`${id}`} className={styles.movieTitleWrapper}>
      <div className={styles.ratingMovie}>
        <p className={styles.rating}>{vote_average}</p>
      </div>

      <div className={styles.movieCardWrapper}>
        <div className={styles.imgCardWrapper}>
          {poster_path ? (
            <img src={urlImg} height="100%" alt={title} />
          ) : (
            <img src={img} width={imgWidth} alt={title} />
          )}
        </div>
      </div>
      <div className={styles.infoWrapper}>
        <MovieTitleInfo heading={titleText} infoField={title} />
        <MovieTitleInfo heading={overviewText} infoField={overview} />
        <MovieTitleInfo heading={releaseText} infoField={release_date} />
        <MovieTitleInfo
          heading={revenueText}
          infoField={`$ ${getRevenueFormat(revenue)}`}
        />
        <MovieTitleInfo
          heading={durationText}
          infoField={`${runtime} ${durationTimeMin}`}
        />
        <div className={styles.genreWrapper}>
          {genres
            ? genres.map((el) => {
                return (
                  <div key={el.id} className={styles.genre}>
                    <span>{`${el.name}`}</span>
                  </div>
                );
              })
            : null}
        </div>
      </div>
    </div>
  );
};

export default MovieTitleCard;
