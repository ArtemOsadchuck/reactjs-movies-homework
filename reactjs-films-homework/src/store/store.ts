import rootStore from './rootStore/rootStore';

export type RootState = ReturnType<typeof rootStore.getState>;
export type AppDispatch = typeof rootStore.dispatch;
export default rootStore;
