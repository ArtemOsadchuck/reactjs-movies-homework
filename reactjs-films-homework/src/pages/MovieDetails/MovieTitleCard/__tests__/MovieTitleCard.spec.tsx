import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';

import MovieTitleCard from '../MovieTitleCard';
import MovieTitleCardMocks from '../mocks';

import { Provider } from 'react-redux';
import store from '../../../../store/store';

describe('MovieTitleCard', () => {
  let fragment: any;

  beforeEach(() => {
    const { asFragment } = render(
      <Provider store={store}>
        <MovieTitleCard props={MovieTitleCardMocks} />{' '}
      </Provider>
    );
    fragment = asFragment();
  });

  afterEach(() => cleanup());

  it('MovieTitleCard snapshot', () => {
    expect(fragment).toMatchSnapshot();
  });

  it('MovieTitleCard img must have attribute:', () => {
    expect(screen.getByRole(/img/i)).toHaveAttribute('src');
    expect(screen.getByRole(/img/i)).toHaveAttribute('height');
    expect(screen.getByRole(/img/i)).toHaveAttribute('alt');
  });

  it('MovieTitleCard heading must have:', () => {
    expect(screen.getByRole(/heading/i)).toHaveClass('title');
    expect(screen.getByRole(/heading/i)).toHaveTextContent(
      'The Shawshank Redemption'
    );
  });

  it('MovieTitleCard must be in the document:', () => {
    expect(screen.getByText(/Overview/i)).toBeInTheDocument();
    expect(screen.getByText(/1994-09-23/i)).toBeInTheDocument();
    expect(screen.getByText(/Revenue/i)).toBeInTheDocument();
    expect(screen.getByText(/341/i)).toBeInTheDocument();
    expect(screen.getByText(/min/i)).toBeInTheDocument();
    expect(screen.getByText(/Drama/i)).toBeInTheDocument();
    expect(screen.getByText(/Crime/i)).toBeInTheDocument();
  });
});
