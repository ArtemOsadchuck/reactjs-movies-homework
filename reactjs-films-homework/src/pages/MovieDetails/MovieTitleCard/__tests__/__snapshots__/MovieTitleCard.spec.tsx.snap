// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTitleCard MovieTitleCard snapshot 1`] = `
<DocumentFragment>
  <div
    class="movieTitleWrapper"
    id="278"
  >
    <div
      class="ratingMovie"
    >
      <p
        class="rating"
      >
        8.7
      </p>
    </div>
    <div
      class="movieCardWrapper"
    >
      <div
        class="imgCardWrapper"
      >
        <img
          alt="The Shawshank Redemption"
          height="100%"
          src="https://image.tmdb.org/t/p/w500//q6y0Go1tsGEsmtFryDOJo3dEmqu.jpg"
        />
      </div>
    </div>
    <div
      class="infoWrapper"
    >
      <div
        class="titleWrapper"
      >
        <p
          class="heading"
        >
          Title:
        </p>
        <h2
          class="title"
        >
          The Shawshank Redemption
        </h2>
      </div>
      <div
        class="otherInfoWrapper"
      >
        <p
          class="heading"
        >
          Overview:
        </p>
        <p
          class="infoField"
        >
          Framed in the 1940s for the double murder of his wife and her lover, upstanding banker Andy Dufresne begins a new life at the Shawshank prison, where he puts his accounting skills to work for an amoral warden. During his long stretch in prison, Dufresne comes to be admired by the other inmates -- including an older prisoner named Red -- for his integrity and unquenchable sense of hope.
        </p>
      </div>
      <div
        class="otherInfoWrapper"
      >
        <p
          class="heading"
        >
          Release date:
        </p>
        <p
          class="infoField"
        >
          1994-09-23
        </p>
      </div>
      <div
        class="otherInfoWrapper"
      >
        <p
          class="heading"
        >
          Revenue:
        </p>
        <p
          class="infoField"
        >
          $ 28 341 469
        </p>
      </div>
      <div
        class="otherInfoWrapper"
      >
        <p
          class="heading"
        >
          Duration:
        </p>
        <p
          class="infoField"
        >
          142 min
        </p>
      </div>
      <div
        class="genreWrapper"
      >
        <div
          class="genre"
        >
          <span>
            Drama
          </span>
        </div>
        <div
          class="genre"
        >
          <span>
            Crime
          </span>
        </div>
      </div>
    </div>
  </div>
   
</DocumentFragment>
`;
